src/messaging/ReliableMessageMgr.cpp

```diff
@@ -187,11 +187,49 @@ CHIP_ERROR ReliableMessageMgr::AddToRetransTable(ReliableMessageContext * rc, Re
     return CHIP_NO_ERROR;
 }
 
+System::Clock::Timestamp ReliableMessageMgr::GetBackoff(System::Clock::Timestamp backoffBase, uint8_t sendCount)
+{
+    static constexpr uint32_t MRP_BACKOFF_JITTER_BASE      = 1000;
+    static constexpr uint32_t MRP_BACKOFF_JITTER_MAX       = 250;
+    static constexpr uint32_t MRP_BACKOFF_BASE_NUMERATOR   = 16;
+    static constexpr uint32_t MRP_BACKOFF_BASE_DENOMENATOR = 10;
+    static constexpr uint32_t MRP_BACKOFF_THRESHOLD        = 1;
+
+    System::Clock::Timestamp backoff = backoffBase;
+
+    // Implement `t = i⋅MRP_BACKOFF_BASE^max(0,n−MRP_BACKOFF_THRESHOLD)` from Section 4.11.2.1. Retransmissions
+
+    // Generate fixed point equivalent of `retryCount = max(0,n−MRP_BACKOFF_THRESHOLD)`
+    int retryCount = sendCount - MRP_BACKOFF_THRESHOLD;
+    if (retryCount < 0)
+        retryCount = 0; // Enforce floor
+    if (retryCount > 4)
+        retryCount = 4; // Enforce reasonable maximum after 5 tries
+
+    // Generate fixed point equivalent of `backoff = i⋅1.6^retryCount`
+    uint32_t backoffNum   = 1;
+    uint32_t backoffDenom = 1;
+    for (int i = 0; i < retryCount; i++)
+    {
+        backoffNum *= MRP_BACKOFF_BASE_NUMERATOR;
+        backoffDenom *= MRP_BACKOFF_BASE_DENOMENATOR;
+    }
+    backoff = backoff * backoffNum / backoffDenom;
+
+    // Implement jitter scaler: `t *= (1.0+random(0,1)⋅MRP_BACKOFF_JITTER)`
+    // where jitter is random multiplier from 1.000 to 1.250:
+    uint32_t jitter = MRP_BACKOFF_JITTER_BASE + Crypto::GetRandU16() % MRP_BACKOFF_JITTER_MAX;
+    backoff         = backoff * jitter / MRP_BACKOFF_JITTER_BASE;
+
+    return backoff;
+}
+
 void ReliableMessageMgr::StartRetransmision(RetransTableEntry * entry)
 {
-    // TODO: Choose active/idle timeout corresponding to the activity of exchanges of the session.
-    entry->nextRetransTime =
-        System::SystemClock().GetMonotonicTimestamp() + entry->ec->GetSessionHandle()->GetMRPConfig().mIdleRetransTimeout;
+    // TODO(#15800): Choose active/idle timeout corresponding to the ActiveState of peer in session.
+    System::Clock::Timestamp backoff =
+        ReliableMessageMgr::GetBackoff(entry->ec->GetSessionHandle()->GetMRPConfig().mActiveRetransTimeout, entry->sendCount);
```

Comment:
Why the change from idle to active here?

Reply:
I don't think this is conformant to proper sleepy or powered device behavior.  Why would we declare our connection is active if we've never heard back from the peer?

Reply:
The current setup (before this change) is that we use idle for the first retransmit and active for subsequent retransmits, fwiw.  I think it's worth keeping that in place in this PR and making changes to that behavior in a separate PR, just as risk-mitigation.

Reply:
Yeah, I know and it baffles me that we currently do this.  It is the opposite of back-off.  But avoiding scope creep is fine.

Reply:
Updated `ExecuteAction` and kept use of active there.  #15800 is tracking fixing idle/active tracking of peers by session.